pkg/util/clientauthorizer/arm.go

```diff
@@ -101,7 +103,13 @@ func (a *arm) refresh() {
 func (a *arm) refreshOnce() error {
 	now := a.now()
 
-	req, err := http.NewRequest(http.MethodGet, strings.TrimSuffix(a.im.Environment().ResourceManagerEndpoint, "/")+":24582/metadata/authentication?api-version=2015-01-01", nil)
+	// ARM <-> RP Authentication endpoint is not consistent.  Check ARM wiki for up-to-date metadata endpoints
+	endpoint := strings.TrimSuffix(a.im.Environment().ResourceManagerEndpoint, "/") + ":24582"
+	if reflect.DeepEqual(a.im.Environment().Environment, azure.PublicCloud) {
```

Comment:
`reflect.DeepEqual` is probably a bit overkill here.  I could probably just compare the name, but all the fields should be populated throughout.